pkg/loop/internal/core/services/capability/capabilities_registry.go

```diff
@@ -264,6 +264,37 @@ func (cr *capabilitiesRegistryClient) Add(ctx context.Context, c capabilities.Ba
 	return nil
 }
 
+func (cr *capabilitiesRegistryClient) Remove(ctx context.Context, c capabilities.BaseCapability) error {
+	info, err := c.Info(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Check the capability and the CapabilityType match here as the ServeNew method does not return an error
+	err = validateCapability(c, info.CapabilityType)
+	if err != nil {
+		return err
+	}
+
+	var cRes net.Resource
+	id, cRes, err := cr.ServeNew(info.ID, func(s *grpc.Server) {
+		pbRegisterCapability(s, cr.BrokerExt, c, info.CapabilityType)
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = cr.grpc.Remove(ctx, &pb.RemoveRequest{
+		CapabilityID: id,
+		Type:         pb.ExecuteAPIType(getExecuteAPIType(info.CapabilityType)),
```

Comment:
I also don't think you need this: Remove should just accept that Base capability interface (the one that exposes info) since all it needs to Remove something is the id.